diff -up signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/common-project-config.pri.cxxflags signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/common-project-config.pri
--- signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/common-project-config.pri.cxxflags	2018-09-17 11:28:27.000000000 -0500
+++ signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/common-project-config.pri	2020-11-08 12:40:44.453972891 -0600
@@ -12,9 +12,9 @@ RCC_DIR         = resources
 #UI_SOURCES_DIR  = ui/src
 
 # we don't like warnings...
-QMAKE_CXXFLAGS -= -Werror -Wno-write-strings
+#QMAKE_CXXFLAGS -= -Werror -Wno-write-strings
 # Disable RTTI
-QMAKE_CXXFLAGS += -fno-rtti
+#QMAKE_CXXFLAGS += -fno-rtti
 # Disable exceptions
 CONFIG += exceptions_off
 # Use C++11
diff -up signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/extensions/cryptsetup/cryptsetup.pro.cxxflags signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/extensions/cryptsetup/cryptsetup.pro
--- signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/extensions/cryptsetup/cryptsetup.pro.cxxflags	2018-09-17 11:28:27.000000000 -0500
+++ signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/extensions/cryptsetup/cryptsetup.pro	2020-11-08 12:49:03.480359607 -0600
@@ -28,8 +28,6 @@ QT += core
 QT -= gui
 
 QMAKE_CXXFLAGS += \
-    -fno-exceptions \
-    -fno-rtti \
     -fvisibility=hidden
 
 DEFINES += QT_NO_CAST_TO_ASCII QT_NO_CAST_FROM_ASCII
diff -up signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/plugins/plugins.pri.cxxflags signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/plugins/plugins.pri
--- signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/plugins/plugins.pri.cxxflags	2018-09-17 11:28:27.000000000 -0500
+++ signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/plugins/plugins.pri	2020-11-08 12:49:27.223473165 -0600
@@ -22,8 +22,8 @@ LIBS += -lsignon-plugins
 QMAKE_LIBDIR += \
     $${TOP_BUILD_DIR}/lib/plugins
 
-QMAKE_CXXFLAGS += -fno-exceptions \
-    -fno-rtti
+QMAKE_CXXFLAGS += -fno-exceptions
+#    -fno-rtti
 
 headers.path = $${INSTALL_PREFIX}/include/signon-plugins
 
diff -up signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/remotepluginprocess/remotepluginprocess.pro.cxxflags signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/remotepluginprocess/remotepluginprocess.pro
--- signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/remotepluginprocess/remotepluginprocess.pro.cxxflags	2018-09-17 11:28:27.000000000 -0500
+++ signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/src/remotepluginprocess/remotepluginprocess.pro	2020-11-08 12:49:44.352555089 -0600
@@ -40,8 +40,8 @@ LIBS += \
     -lsignon-plugins-common \
     -lsignon-plugins
 
-QMAKE_CXXFLAGS += -fno-exceptions \
-                  -fno-rtti
+#QMAKE_CXXFLAGS += -fno-exceptions \
+#                  -fno-rtti
 
 #DEFINES += QT_NO_CAST_TO_ASCII \
 #    QT_NO_CAST_FROM_ASCII
