diff -up signon-8.57/lib/SignOn/SignOn.pro.no_static signon-8.57/lib/SignOn/SignOn.pro
--- signon-8.57/lib/SignOn/SignOn.pro.no_static	2014-08-25 09:33:29.000000000 -0500
+++ signon-8.57/lib/SignOn/SignOn.pro	2015-04-01 06:57:53.337391324 -0500
@@ -1,7 +1,7 @@
 TEMPLATE = subdirs
 CONFIG   += ordered
 SUBDIRS = \
-    libsignon-qt-shared.pro \
-    libsignon-qt-static.pro
+    libsignon-qt-shared.pro
+#   libsignon-qt-static.pro
 
 include(doc/doc.pri)
