diff --git a/common-project-config.pri b/common-project-config.pri
index 3206471..5ba0269 100644
--- a/common-project-config.pri
+++ b/common-project-config.pri
@@ -7,7 +7,7 @@ CONFIG         += link_pkgconfig
 # we don't like warnings...
 QMAKE_CXXFLAGS -= -Werror -Wno-write-strings
 # Disable RTTI
-QMAKE_CXXFLAGS += -fno-exceptions -fno-rtti
+# QMAKE_CXXFLAGS += -fno-exceptions -fno-rtti
 
 TOP_SRC_DIR     = $$PWD
 TOP_BUILD_DIR   = $${TOP_SRC_DIR}/$(BUILD_DIR)
diff --git a/tests/unit/tst_inactivity_timer.pro b/tests/unit/tst_inactivity_timer.pro
index 1b80250..072a994 100644
--- a/tests/unit/tst_inactivity_timer.pro
+++ b/tests/unit/tst_inactivity_timer.pro
@@ -23,9 +23,9 @@ INCLUDEPATH += \
     . \
     $$TOP_SRC_DIR/src
 
-QMAKE_CXXFLAGS += \
-    -fno-exceptions \
-    -fno-rtti
+#QMAKE_CXXFLAGS += \
+#    -fno-exceptions \
+#    -fno-rtti
 
 DEFINES += \
     DEBUG_ENABLED \
diff --git a/tests/unit/tst_signon_ui.pro b/tests/unit/tst_signon_ui.pro
index 815e52b..155e327 100644
--- a/tests/unit/tst_signon_ui.pro
+++ b/tests/unit/tst_signon_ui.pro
@@ -74,9 +74,9 @@ INCLUDEPATH += \
     $$TOP_SRC_DIR/src \
     $$OUT_PWD/../../src
 
-QMAKE_CXXFLAGS += \
-    -fno-exceptions \
-    -fno-rtti
+#QMAKE_CXXFLAGS += \
+#    -fno-exceptions \
+#    -fno-rtti
 
 DEFINES += \
     DEBUG_ENABLED \
