diff --git a/CMakeLists.txt b/CMakeLists.txt
index 979d263a9..15e4ddd3f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,8 +8,8 @@ set(RELEASE_SERVICE_VERSION "${RELEASE_SERVICE_VERSION_MAJOR}.${RELEASE_SERVICE_
 
 project(KDialog VERSION ${RELEASE_SERVICE_VERSION})
 
-set(QT_MIN_VERSION "6.7.0")
-set(KF_MIN_VERSION "6.3.0")
+set(QT_MIN_VERSION "6.5.0")
+set(KF_MIN_VERSION "5.240.0")
 
 # ECM setup
 find_package(ECM ${KF_MIN_VERSION} CONFIG REQUIRED)
@@ -33,7 +33,7 @@ if(NOT WITHOUT_X11)
 endif()
 
 ecm_set_disabled_deprecation_versions(
-    QT 6.7.0
+    QT 6.5.0
     KF 6.4.0
 )
 
