diff --git a/CMakeLists.txt b/CMakeLists.txt
index 30fb5ac..3e381e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,8 +12,8 @@ set(MARKDOWNPART_VERSION "${MARKDOWNPART_BASE_VERSION}.${RELEASE_SERVICE_COMPACT
 
 project(markdownpart VERSION ${MARKDOWNPART_VERSION})
 
-set(QT_MIN_VERSION "6.7.0")
-set(KF_MIN_VERSION "6.3.0")
+set(QT_MIN_VERSION "6.5.0")
+set(KF_MIN_VERSION "6.0.0")
 
 find_package(ECM ${KF_MIN_VERSION} REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
@@ -33,8 +33,8 @@ find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS
 )
 
 ecm_set_disabled_deprecation_versions(
-    QT 6.7
-    KF 6.4
+    QT 6.6
+    KF 6.0
 )
 add_definitions(-DQT_NO_CONTEXTLESS_CONNECT)
 add_subdirectory(src)
