diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb2c576..f8e0bb6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,12 +94,13 @@ find_package(KF${QT_MAJOR_VERSION} ${KF_MIN_VERSION} REQUIRED COMPONENTS
     WidgetsAddons
 )
 find_package(KF${QT_MAJOR_VERSION} ${KF_MIN_VERSION} OPTIONAL_COMPONENTS
-    Activities
     DocTools
     Purpose
 )
 
-set(HAVE_KACTIVITIES ${KF${QT_MAJOR_VERSION}Activities_FOUND})
+find_package(PlasmaActivities)
+
+set(HAVE_KACTIVITIES ${PlasmaActivities_FOUND})
 set(HAVE_PURPOSE ${KF${QT_MAJOR_VERSION}Purpose_FOUND})
 
 ecm_set_disabled_deprecation_versions(QT 5.14.0 KF 5.99.0
diff --git a/app/CMakeLists.txt b/app/CMakeLists.txt
index afeb708..ce9ed65 100644
--- a/app/CMakeLists.txt
+++ b/app/CMakeLists.txt
@@ -103,8 +103,8 @@ target_link_libraries(gwenview
     gwenviewlib
     )
 
-if (KF${QT_MAJOR_VERSION}Activities_FOUND)
-    target_link_libraries(gwenview KF${QT_MAJOR_VERSION}::Activities)
+if (PlasmaActivities_FOUND)
+    target_link_libraries(gwenview Plasma::Activities)
 endif()
 
 if (KF${QT_MAJOR_VERSION}Purpose_FOUND)
diff --git a/app/viewmainpage.cpp b/app/viewmainpage.cpp
index fc1e9e0..c967127 100644
--- a/app/viewmainpage.cpp
+++ b/app/viewmainpage.cpp
@@ -37,7 +37,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #include <KSqueezedTextLabel>
 #include <KToggleAction>
 #if HAVE_KACTIVITIES
-#include <KActivities/ResourceInstance>
+#include <PlasmaActivities/ResourceInstance>
 #endif
 
 // Local
