diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 70fc45b8f..8c6f484cc 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -182,7 +182,7 @@ ecm_generate_export_header(KF6KIOCore
     USE_VERSION_HEADER
     VERSION_BASE_NAME KIO
     DEPRECATED_BASE_VERSION 0
-    DEPRECATION_VERSIONS 5.101 6.3 6.4 6.6 6.9 6.11 6.15
+    DEPRECATION_VERSIONS 5.101 6.3 6.4 6.6 6.9 6.11
     EXCLUDE_DEPRECATED_BEFORE_AND_AT ${EXCLUDE_DEPRECATED_BEFORE_AND_AT}
 )
 
diff --git a/src/core/jobuidelegateextension.h b/src/core/jobuidelegateextension.h
index d99cc4f5f..0ce7656c4 100644
--- a/src/core/jobuidelegateextension.h
+++ b/src/core/jobuidelegateextension.h
@@ -126,7 +126,6 @@ protected:
     virtual ~JobUiDelegateExtension();
 
 public:
-#if KIOCORE_ENABLE_DEPRECATED_SINCE(6, 15)
     /*!
      * The type of deletion: real deletion, moving the files to the trash
      * or emptying the trash
@@ -171,9 +170,7 @@ public:
      *
      * Returns \c true if confirmed
      */
-    KIOCORE_DEPRECATED_VERSION(6, 15, "Use AskUserActionInterface::askUserDelete instead")
     virtual bool askDeleteConfirmation(const QList<QUrl> &urls, DeletionType deletionType, ConfirmationType confirmationType) = 0;
-#endif
 
     /*!
      * \value UpdateContent
diff --git a/src/filewidgets/CMakeLists.txt b/src/filewidgets/CMakeLists.txt
index 0bf97b356..73f032990 100644
--- a/src/filewidgets/CMakeLists.txt
+++ b/src/filewidgets/CMakeLists.txt
@@ -97,7 +97,7 @@ ecm_generate_export_header(KF6KIOFileWidgets
     USE_VERSION_HEADER
     VERSION_BASE_NAME KIO
     DEPRECATED_BASE_VERSION 0
-    DEPRECATION_VERSIONS 6.3 6.15
+    DEPRECATION_VERSIONS 6.3
     EXCLUDE_DEPRECATED_BEFORE_AND_AT ${EXCLUDE_DEPRECATED_BEFORE_AND_AT}
 )
 
diff --git a/src/filewidgets/kdiroperator.cpp b/src/filewidgets/kdiroperator.cpp
index bfd241a1d..4a2e400c7 100644
--- a/src/filewidgets/kdiroperator.cpp
+++ b/src/filewidgets/kdiroperator.cpp
@@ -40,14 +40,11 @@
 #include <KUrlMimeData>
 #include <kfileitemdelegate.h>
 #include <kfilepreviewgenerator.h>
-#include <kio/askuseractioninterface.h>
 #include <kio/copyjob.h>
 #include <kio/deletejob.h>
 #include <kio/deleteortrashjob.h>
 #include <kio/jobuidelegate.h>
-#include <kio/jobuidelegatefactory.h>
 #include <kio/previewjob.h>
-#include <kio/widgetsaskuseractionhandler.h>
 #include <kpropertiesdialog.h>
 
 #include <QActionGroup>
@@ -715,7 +712,6 @@ void KDirOperator::mkdir()
     d->m_newFileMenu->createDirectory();
 }
 
-#if KIOWIDGETS_BUILD_DEPRECATED_SINCE(6, 15)
 KIO::DeleteJob *KDirOperator::del(const KFileItemList &items, QWidget *parent, bool ask, bool showProgress)
 {
     if (items.isEmpty()) {
@@ -746,7 +742,6 @@ KIO::DeleteJob *KDirOperator::del(const KFileItemList &items, QWidget *parent, b
 
     return nullptr;
 }
-#endif
 
 void KDirOperator::deleteSelected()
 {
@@ -761,7 +756,6 @@ void KDirOperator::deleteSelected()
     deleteJob->start();
 }
 
-#if KIOWIDGETS_BUILD_DEPRECATED_SINCE(6, 15)
 KIO::CopyJob *KDirOperator::trash(const KFileItemList &items, QWidget *parent, bool ask, bool showProgress)
 {
     if (items.isEmpty()) {
@@ -788,7 +782,6 @@ KIO::CopyJob *KDirOperator::trash(const KFileItemList &items, QWidget *parent, b
 
     return nullptr;
 }
-#endif
 
 KFilePreviewGenerator *KDirOperator::previewGenerator() const
 {
diff --git a/src/filewidgets/kdiroperator.h b/src/filewidgets/kdiroperator.h
index 11e8d0508..87ce4e211 100644
--- a/src/filewidgets/kdiroperator.h
+++ b/src/filewidgets/kdiroperator.h
@@ -568,7 +568,6 @@ public:
      */
     bool followSelectedDirectories() const;
 
-#if KIOFILEWIDGETS_ENABLE_DEPRECATED_SINCE(6, 15)
     /*!
      * Starts and returns a KIO::DeleteJob to delete the given \a items.
      *
@@ -579,11 +578,8 @@ public:
      * \a ask specifies whether a confirmation dialog should be shown
      *
      * \a showProgress passed to the DeleteJob to show a progress dialog
-     * \deprecated[6.15]
      */
-    KIOFILEWIDGETS_DEPRECATED_VERSION(6, 15, "Use KIO::DeleteOrTrashJob instead")
     virtual KIO::DeleteJob *del(const KFileItemList &items, QWidget *parent = nullptr, bool ask = true, bool showProgress = true);
-#endif
 
     /*!
      * Clears the forward and backward history.
@@ -645,7 +641,6 @@ public:
      */
     virtual void setDropOptions(int options);
 
-#if KIOFILEWIDGETS_ENABLE_DEPRECATED_SINCE(6, 15)
     /*!
      * Starts and returns a KIO::CopyJob to trash the given \a items.
      *
@@ -654,9 +649,7 @@ public:
      * \a ask specifies whether a confirmation dialog should be shown
      * \a showProgress passed to the CopyJob to show a progress dialog
      */
-    KIOFILEWIDGETS_DEPRECATED_VERSION(6, 15, "Use KIO::DeleteOrTrashJob instead")
     virtual KIO::CopyJob *trash(const KFileItemList &items, QWidget *parent, bool ask = true, bool showProgress = true);
-#endif
 
     /*!
      * Returns the preview generator for the current view.
diff --git a/src/widgets/jobuidelegate.cpp b/src/widgets/jobuidelegate.cpp
index 5304f6e28..a032fec6f 100644
--- a/src/widgets/jobuidelegate.cpp
+++ b/src/widgets/jobuidelegate.cpp
@@ -175,7 +175,6 @@ void KIO::JobUiDelegate::unregisterWindow(QWidget *window)
     s_static()->slotUnregisterWindow(window);
 }
 
-#if KIOWIDGETS_BUILD_DEPRECATED_SINCE(6, 15)
 bool KIO::JobUiDelegate::askDeleteConfirmation(const QList<QUrl> &urls, DeletionType deletionType, ConfirmationType confirmationType)
 {
     QString keyName;
@@ -301,7 +300,6 @@ bool KIO::JobUiDelegate::askDeleteConfirmation(const QList<QUrl> &urls, Deletion
     }
     return true;
 }
-#endif
 
 KIO::ClipboardUpdater *KIO::JobUiDelegate::createClipboardUpdater(Job *job, ClipboardUpdaterMode mode)
 {
diff --git a/src/widgets/jobuidelegate.h b/src/widgets/jobuidelegate.h
index b9d89be47..c3ced4b8d 100644
--- a/src/widgets/jobuidelegate.h
+++ b/src/widgets/jobuidelegate.h
@@ -12,7 +12,6 @@
 #ifndef KIO_JOBUIDELEGATE_H
 #define KIO_JOBUIDELEGATE_H
 
-#include "kiowidgets_export.h"
 #include <KDialogJobUiDelegate>
 #include <kio/askuseractioninterface.h>
 #include <kio/global.h>
@@ -86,7 +85,6 @@ public:
      */
     static void unregisterWindow(QWidget *window);
 
-#if KIOWIDGETS_ENABLE_DEPRECATED_SINCE(6, 15)
     /*!
      * Ask for confirmation before deleting/trashing \a urls.
      *
@@ -102,11 +100,8 @@ public:
      * \note The window passed to setWindow is used as the parent for the message box.
      *
      * Returns true if confirmed
-     *
-     * \deprecated[6.15] Use AskUserActionInterface::askUserDelete
      */
     bool askDeleteConfirmation(const QList<QUrl> &urls, DeletionType deletionType, ConfirmationType confirmationType) override;
-#endif
 
     /*!
      * Creates a clipboard updater
