diff --git a/src/globalshortcuts.cpp b/src/globalshortcuts.cpp
index be1ac38ab5..b42fc0db00 100644
--- a/src/globalshortcuts.cpp
+++ b/src/globalshortcuts.cpp
@@ -106,6 +106,7 @@ void GlobalShortcutsManager::init()
             qCDebug(KWIN_CORE) << "Init of kglobalaccel failed";
             m_kglobalAccel.reset();
         } else {
+            m_kglobalAccelInterface = m_kglobalAccel->interface();
             qCDebug(KWIN_CORE) << "KGlobalAcceld inited";
         }
     }
diff --git a/src/globalshortcuts.h b/src/globalshortcuts.h
index 29141e6017..b577ed5c1a 100644
--- a/src/globalshortcuts.h
+++ b/src/globalshortcuts.h
@@ -111,13 +111,6 @@ public:
     void processPinchCancel();
     void processPinchEnd();
 
-#if KWIN_BUILD_GLOBALSHORTCUTS
-    void setKGlobalAccelInterface(KGlobalAccelInterface *interface)
-    {
-        m_kglobalAccelInterface = interface;
-    }
-#endif
-
 private:
     void objectDeleted(QObject *object);
     bool add(GlobalShortcut sc, DeviceType device = DeviceType::Touchpad);
diff --git a/src/input.cpp b/src/input.cpp
index d7cf64b2d6..c40f998a81 100644
--- a/src/input.cpp
+++ b/src/input.cpp
@@ -3431,13 +3431,6 @@ void InputRedirection::registerTouchpadPinchShortcut(PinchDirection direction, u
 #endif
 }
 
-void InputRedirection::registerGlobalAccel(KGlobalAccelInterface *interface)
-{
-#if KWIN_BUILD_GLOBALSHORTCUTS
-    m_shortcuts->setKGlobalAccelInterface(interface);
-#endif
-}
-
 void InputRedirection::registerTouchscreenSwipeShortcut(SwipeDirection direction, uint fingerCount, QAction *action, std::function<void(qreal)> progressCallback)
 {
 #if KWIN_BUILD_GLOBALSHORTCUTS
diff --git a/src/input.h b/src/input.h
index 1cebb442b8..d3ea887211 100644
--- a/src/input.h
+++ b/src/input.h
@@ -92,7 +92,6 @@ public:
     void registerTouchpadPinchShortcut(PinchDirection direction, uint32_t fingerCount, QAction *onUp, std::function<void(qreal)> progressCallback = {});
     void registerTouchscreenSwipeShortcut(SwipeDirection direction, uint32_t fingerCount, QAction *action, std::function<void(qreal)> progressCallback = {});
     void forceRegisterTouchscreenSwipeShortcut(SwipeDirection direction, uint32_t fingerCount, QAction *action, std::function<void(qreal)> progressCallback = {});
-    void registerGlobalAccel(KGlobalAccelInterface *interface);
 
     bool supportsPointerWarping() const;
     void warpPointer(const QPointF &pos);
diff --git a/src/plugins/kglobalaccel/kglobalaccel_plugin.cpp b/src/plugins/kglobalaccel/kglobalaccel_plugin.cpp
index 08b739e4e2..8fa2c790c3 100644
--- a/src/plugins/kglobalaccel/kglobalaccel_plugin.cpp
+++ b/src/plugins/kglobalaccel/kglobalaccel_plugin.cpp
@@ -24,24 +24,6 @@ bool KGlobalAccelImpl::grabKey(int key, bool grab)
     return true;
 }
 
-void KGlobalAccelImpl::setEnabled(bool enabled)
-{
-    if (m_shuttingDown) {
-        return;
-    }
-    static KWin::InputRedirection *s_input = KWin::InputRedirection::self();
-    if (!s_input) {
-        qFatal("This plugin is intended to be used with KWin and this is not KWin, exiting now");
-    } else {
-        if (!m_inputDestroyedConnection) {
-            m_inputDestroyedConnection = connect(s_input, &QObject::destroyed, this, [this] {
-                m_shuttingDown = true;
-            });
-        }
-    }
-    s_input->registerGlobalAccel(enabled ? this : nullptr);
-}
-
 bool KGlobalAccelImpl::checkKeyPressed(int keyQt, KWin::KeyboardKeyState state)
 {
     switch (state) {
diff --git a/src/plugins/kglobalaccel/kglobalaccel_plugin.h b/src/plugins/kglobalaccel/kglobalaccel_plugin.h
index 2776f5533b..64d2df25d6 100644
--- a/src/plugins/kglobalaccel/kglobalaccel_plugin.h
+++ b/src/plugins/kglobalaccel/kglobalaccel_plugin.h
@@ -25,14 +25,9 @@ public:
     ~KGlobalAccelImpl() override;
 
     bool grabKey(int key, bool grab) override;
-    void setEnabled(bool) override;
 
 public Q_SLOTS:
     bool checkKeyPressed(int keyQt, KWin::KeyboardKeyState state);
     bool checkPointerPressed(Qt::MouseButtons buttons);
     bool checkAxisTriggered(int axis);
-
-private:
-    bool m_shuttingDown = false;
-    QMetaObject::Connection m_inputDestroyedConnection;
 };
