diff --git a/src/plugins/kglobalaccel/kglobalaccel_plugin.h b/src/plugins/kglobalaccel/kglobalaccel_plugin.h
index 2776f5533b..22d1ef306b 100644
--- a/src/plugins/kglobalaccel/kglobalaccel_plugin.h
+++ b/src/plugins/kglobalaccel/kglobalaccel_plugin.h
@@ -25,7 +25,6 @@ public:
     ~KGlobalAccelImpl() override;
 
     bool grabKey(int key, bool grab) override;
-    void setEnabled(bool) override;
 
 public Q_SLOTS:
     bool checkKeyPressed(int keyQt, KWin::KeyboardKeyState state);
