diff --git a/src/lib/preferences/preferences.cpp b/src/lib/preferences/preferences.cpp
index 0988cc420..43c3670bf 100644
--- a/src/lib/preferences/preferences.cpp
+++ b/src/lib/preferences/preferences.cpp
@@ -289,7 +289,7 @@ Preferences::Preferences(BrowserWindow* window)
     ui->animateScrolling->setChecked(settings.value(QSL("AnimateScrolling"), true).toBool());
     ui->wheelScroll->setValue(settings.value(QSL("wheelScrollLines"), qApp->wheelScrollLines()).toInt());
     ui->printEBackground->setChecked(settings.value(QSL("PrintElementBackground"), true).toBool());
-    ui->useNativeScrollbars->setChecked(settings.value(QSL("UseNativeScrollbars"), false).toBool());
+    ui->useNativeScrollbars->setChecked(settings.value(QSL("UseNativeScrollbars"), true).toBool());
     ui->disableVideoAutoPlay->setChecked(settings.value(QSL("DisableVideoAutoPlay"), false).toBool());
     ui->webRTCPublicIpOnly->setChecked(settings.value(QSL("WebRTCPublicIpOnly"), true).toBool());
     ui->dnsPrefetch->setChecked(settings.value(QSL("DNSPrefetch"), true).toBool());
diff --git a/src/lib/webengine/webscrollbarmanager.cpp b/src/lib/webengine/webscrollbarmanager.cpp
index ee84f7db6..7e9e70bdd 100644
--- a/src/lib/webengine/webscrollbarmanager.cpp
+++ b/src/lib/webengine/webscrollbarmanager.cpp
@@ -108,7 +108,7 @@ WebScrollBarManager::WebScrollBarManager(QObject *parent)
 
 void WebScrollBarManager::loadSettings()
 {
-    m_enabled = Settings().value(QSL("Web-Browser-Settings/UseNativeScrollbars"), false).toBool();
+    m_enabled = Settings().value(QSL("Web-Browser-Settings/UseNativeScrollbars"), true).toBool();
 
     if (!m_enabled) {
         for (WebView *view : m_scrollbars.keys()) {
