diff -ur kpimtextedit-23.08.1/src/CMakeLists.txt kpimtextedit-23.08.1-install-and-export-for-blogilo/src/CMakeLists.txt
--- kpimtextedit-23.08.1/src/CMakeLists.txt	2023-09-03 06:09:00.000000000 +0200
+++ kpimtextedit-23.08.1-install-and-export-for-blogilo/src/CMakeLists.txt	2023-09-20 06:42:51.719871321 +0200
@@ -202,6 +202,9 @@
 ecm_generate_headers(KPimTextEdit_CamelCase_HEADERS
     HEADER_NAMES
     EditorUtil
+    InsertHtmlDialog
+    InsertImageWidget
+    InsertTableWidget
     SelectSpecialCharDialog
     SlideContainer
     TextEditorCompleter
diff -ur kpimtextedit-23.08.1/src/inserthtmldialog.h kpimtextedit-23.08.1-install-and-export-for-blogilo/src/inserthtmldialog.h
--- kpimtextedit-23.08.1/src/inserthtmldialog.h	2023-09-03 06:09:00.000000000 +0200
+++ kpimtextedit-23.08.1-install-and-export-for-blogilo/src/inserthtmldialog.h	2023-09-20 06:42:51.719871321 +0200
@@ -10,12 +10,12 @@
 #include <QDialog>
 
 #include "kpimtextedit/plaintexteditor.h"
-#include "kpimtextedit_private_export.h"
+#include "kpimtextedit_export.h"
 
 namespace KPIMTextEdit
 {
 class InsertHtmlDialogPrivate;
-class KPIMTEXTEDIT_TESTS_EXPORT InsertHtmlDialog : public QDialog
+class KPIMTEXTEDIT_EXPORT InsertHtmlDialog : public QDialog
 {
     Q_OBJECT
 public:
diff -ur kpimtextedit-23.08.1/src/insertimagewidget.h kpimtextedit-23.08.1-install-and-export-for-blogilo/src/insertimagewidget.h
--- kpimtextedit-23.08.1/src/insertimagewidget.h	2023-09-03 06:09:00.000000000 +0200
+++ kpimtextedit-23.08.1-install-and-export-for-blogilo/src/insertimagewidget.h	2023-09-20 06:42:51.846869477 +0200
@@ -6,6 +6,8 @@
 */
 #pragma once
 
+#include "kpimtextedit_export.h"
+
 #include <QUrl>
 
 #include <QWidget>
@@ -14,7 +16,7 @@
 {
 class InsertImageWidgetPrivate;
 
-class InsertImageWidget : public QWidget
+class KPIMTEXTEDIT_EXPORT InsertImageWidget : public QWidget
 {
     Q_OBJECT
 public:
diff -ur kpimtextedit-23.08.1/src/inserttablewidget.h kpimtextedit-23.08.1-install-and-export-for-blogilo/src/inserttablewidget.h
--- kpimtextedit-23.08.1/src/inserttablewidget.h	2023-09-03 06:09:00.000000000 +0200
+++ kpimtextedit-23.08.1-install-and-export-for-blogilo/src/inserttablewidget.h	2023-09-20 06:42:51.847869462 +0200
@@ -7,13 +7,15 @@
 
 #pragma once
 
+#include "kpimtextedit_export.h"
+
 #include <QWidget>
 
 #include <QTextLength>
 
 namespace KPIMTextEdit
 {
-class InsertTableWidget : public QWidget
+class KPIMTEXTEDIT_EXPORT InsertTableWidget : public QWidget
 {
     Q_OBJECT
 public:
