diff --git a/KTorrent6Config.cmake.in b/KTorrent6Config.cmake.in
index 9ed2c95..104f886 100644
--- a/KTorrent6Config.cmake.in
+++ b/KTorrent6Config.cmake.in
@@ -8,6 +8,7 @@ find_dependency(KF6KIO "@KF_MIN_VERSION@")
 
 find_dependency(Qt6Core "@QT_MIN_VERSION@")
 find_dependency(Qt6Network "@QT_MIN_VERSION@")
+find_dependency(Qt6Core5Compat "@QT_MIN_VERSION@")
 
 find_dependency(Boost "@Boost_MIN_VERSION@")
 find_dependency(LibGMP "@LibGMP_MIN_VERSION@")
