diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0bad7c8..de05fef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,7 @@ if (QT_MAJOR_VERSION STREQUAL "6")
     find_package(KF6StatusNotifierItem ${KF_MIN_VERSION} REQUIRED)
 endif()
 
-find_package(KF${QT_MAJOR_VERSION} ${KF_MIN_VERSION} OPTIONAL_COMPONENTS Activities)
+find_package(KF${QT_MAJOR_VERSION} OPTIONAL_COMPONENTS Activities)
 set_package_properties(KF${QT_MAJOR_VERSION}Activities PROPERTIES DESCRIPTION "KActivities libraries"
                        URL "https://www.kde.org"
                        TYPE OPTIONAL
