diff --git a/modules/ECMGeneratePythonBindings.cmake b/modules/ECMGeneratePythonBindings.cmake
index 147d0c72..27c705cf 100644
--- a/modules/ECMGeneratePythonBindings.cmake
+++ b/modules/ECMGeneratePythonBindings.cmake
@@ -172,7 +172,7 @@ function(ecm_generate_python_bindings)
     add_custom_command(
         TARGET ${PB_PACKAGE_NAME}
         POST_BUILD
-        COMMAND Python3::Interpreter -m build --wheel
+        COMMAND Python3::Interpreter -m build --wheel --no-isolation
         WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${PB_PACKAGE_NAME}"
         COMMENT "Building Python Wheel"
     )
