diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2c39a4c89..f4e41d05e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,11 +8,11 @@ set (RELEASE_SERVICE_VERSION "${RELEASE_SERVICE_VERSION_MAJOR}.${RELEASE_SERVICE
 
 project(kfind VERSION ${RELEASE_SERVICE_VERSION})
 
-set (KF_MIN_VERSION "6.3.0")
+set (KF_MIN_VERSION "5.240.0")
 # ECM setup
 find_package(ECM ${KF_MIN_VERSION} CONFIG REQUIRED)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
-set(QT_REQUIRED_VERSION "6.7.0")
+set(QT_REQUIRED_VERSION "6.5.0")
 include(KDEInstallDirs)
 find_package(Qt6 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Widgets Concurrent)
 find_package(Qt6Core5Compat REQUIRED)
@@ -47,8 +47,8 @@ set_package_properties(KF6DocTools PROPERTIES DESCRIPTION
     "Tools to generate documentation"
     TYPE OPTIONAL
 )
-ecm_set_disabled_deprecation_versions(QT 6.7
-    KF 6.4.0
+ecm_set_disabled_deprecation_versions(QT 6.6
+    KF 6.0.0
     )
 add_definitions(-DQT_NO_CONTEXTLESS_CONNECT)
 add_subdirectory(src)
