diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7626983..2a605b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -190,6 +190,11 @@ if(PipeWire_FOUND)
   find_package(Libdrm REQUIRED)
 endif()
 
+pkg_check_modules(libxcrypt libxcrypt>=4)
+if(${libxcrypt_FOUND})
+    set(HAVE_CRYPT_GENSALT true)
+endif()
+
 find_package(Qt6WaylandClient COMPONENTS Private)
 find_package(PlasmaWaylandProtocols 1.6 REQUIRED)
 find_package(Wayland REQUIRED COMPONENTS Client Server) # Server is used in autotests
diff --git a/config-workspace.h.cmake b/config-workspace.h.cmake
index 74e0943..fdad76c 100644
--- a/config-workspace.h.cmake
+++ b/config-workspace.h.cmake
@@ -6,6 +6,9 @@
 /* Define to 1 if you have the <sys/time.h> header file. */
 #cmakedefine HAVE_SYS_TIME_H 1
 
+/* Define to 1 if you have libxcrypt with crypt_gensalt. */
+#cmakedefine01 HAVE_CRYPT_GENSALT
+
 /* place where plasma-frameworks things are installed */
 #define PLASMA_RELATIVE_DATA_INSTALL_DIR "@PLASMA_RELATIVE_DATA_INSTALL_DIR@"
 
diff --git a/kcms/users/src/user.cpp b/kcms/users/src/user.cpp
index 1d82eb3..4a73e42 100644
--- a/kcms/users/src/user.cpp
+++ b/kcms/users/src/user.cpp
@@ -9,6 +9,7 @@
 #include "user.h"
 #include "kcmusers_debug.h"
 #include "user_interface.h"
+#include <config-workspace.h>
 #include <KLocalizedString>
 #include <KWallet>
 #include <QImage>
@@ -16,6 +17,10 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+#if HAVE_CRYPT_GENSALT
+#include <crypt.h>
+#endif
+
 User::User(QObject *parent)
     : QObject(parent)
 {
@@ -197,6 +202,7 @@ void User::loadData()
     }
 }
 
+#if !(HAVE_CRYPT_GENSALT)
 static char saltCharacter()
 {
     static constexpr const quint32 letterCount = 64;
@@ -211,9 +217,13 @@ static char saltCharacter()
 
     return saltCharacters[index];
 }
+#endif
 
 static QString saltPassword(const QString &plain)
 {
+#if HAVE_CRYPT_GENSALT
+    QString salt = crypt_gensalt (NULL, 0, NULL, 0);
+#else
     QString salt;
 
     salt.append("$6$");
@@ -223,6 +233,7 @@ static QString saltPassword(const QString &plain)
     }
 
     salt.append("$");
+#endif
 
     auto stdStrPlain = plain.toStdString();
     auto cStrPlain = stdStrPlain.c_str();
