diff -up cantor-24.02.2/src/backends/julia/juliaserver/juliaserver.cpp.me cantor-24.02.2/src/backends/julia/juliaserver/juliaserver.cpp
--- cantor-24.02.2/src/backends/julia/juliaserver/juliaserver.cpp.me	2024-05-06 14:32:14.389234985 +0200
+++ cantor-24.02.2/src/backends/julia/juliaserver/juliaserver.cpp	2024-05-06 14:33:18.398528912 +0200
@@ -168,7 +168,7 @@ void JuliaServer::parseJlModule(jl_modul
 
     jl_function_t* jl_names_function = jl_get_function(jl_base_module, "names");
     jl_value_t* names = jl_call1(jl_names_function, (jl_value_t*)module);
-    jl_value_t **data = (jl_value_t**)jl_array_data(names);
+    jl_value_t **data = (jl_value_t**)jl_array_data_(names);
     for (size_t i = 0; i < jl_array_len(names); i++)
     {
         bool isBindingResolved = (bool)jl_binding_resolved_p(module, (jl_sym_t*)(data[i]));
