diff --git a/src/ShapeCornersShader.cpp b/src/ShapeCornersShader.cpp
index f7801c9..8044e80 100644
--- a/src/ShapeCornersShader.cpp
+++ b/src/ShapeCornersShader.cpp
@@ -97,7 +97,7 @@ ShapeCornersShader::Bind(KWin::EffectWindow *w, qreal scale) const {
 const std::unique_ptr<KWin::GLShader>&
 ShapeCornersShader::Bind(const QMatrix4x4& mvp, KWin::EffectWindow *w, qreal scale) const {
     Bind(w, scale);
-    m_shader->setUniform(KWin::GLShader::ModelViewProjectionMatrix, mvp);
+    m_shader->setUniform(KWin::GLShader::Mat4Uniform::ModelViewProjectionMatrix, mvp);
     return m_shader;
 }
 
