diff --git a/llm/generate/gen_common.sh b/llm/generate/gen_common.sh
index 1186a06..fb49580 100644
--- a/llm/generate/gen_common.sh
+++ b/llm/generate/gen_common.sh
@@ -54,8 +54,6 @@ git_module_setup() {
         echo "Cleaning up old submodule"
         rm -rf ${LLAMACPP_DIR}
     fi
-    git submodule init
-    git submodule update --force ${LLAMACPP_DIR}
 
 }
 
@@ -68,29 +66,23 @@ apply_patches() {
     if [ -n "$(ls -A ../patches/*.diff)" ]; then
         # apply temporary patches until fix is upstream
         for patch in ../patches/*.diff; do
-            for file in $(grep "^+++ " ${patch} | cut -f2 -d' ' | cut -f2- -d/); do
-                (cd ${LLAMACPP_DIR}; git checkout ${file})
-            done
-        done
-        for patch in ../patches/*.diff; do
-            (cd ${LLAMACPP_DIR} && git apply ${patch})
+            (cd ${LLAMACPP_DIR} && patch -Np1 -i ${patch})
         done
     fi
 }
 
 build() {
     cmake -S ${LLAMACPP_DIR} -B ${BUILD_DIR} ${CMAKE_DEFS}
-    cmake --build ${BUILD_DIR} ${CMAKE_TARGETS} -j8
+    cmake --build ${BUILD_DIR} ${CMAKE_TARGETS} -j${RPM_BUILD_NCPUS} --verbose
     mkdir -p ${BUILD_DIR}/lib/
-    ls ${BUILD_DIR}
-    g++ -fPIC -g -shared -o ${BUILD_DIR}/lib/libext_server.${LIB_EXT} \
+    g++ -fPIC ${CXXFLAGS} -shared ${LDFLAGS} -o ${BUILD_DIR}/lib/libext_server.${LIB_EXT} \
         ${GCC_ARCH} \
         ${WHOLE_ARCHIVE} ${BUILD_DIR}/ext_server/libext_server.a ${NO_WHOLE_ARCHIVE} \
         ${BUILD_DIR}/common/libcommon.a \
         ${BUILD_DIR}/libllama.a \
-        -Wl,-rpath,\$ORIGIN \
         -lpthread -ldl -lm \
         ${EXTRA_LIBS}
+    strip --strip-all ${BUILD_DIR}/lib/libext_server.${LIB_EXT}
 }
 
 compress_libs() {
@@ -110,13 +102,5 @@ compress_libs() {
 
 # Keep the local tree clean after we're done with the build
 cleanup() {
-    (cd ${LLAMACPP_DIR}/ && git checkout CMakeLists.txt)
-
-    if [ -n "$(ls -A ../patches/*.diff)" ]; then
-        for patch in ../patches/*.diff; do
-            for file in $(grep "^+++ " ${patch} | cut -f2 -d' ' | cut -f2- -d/); do
-                (cd ${LLAMACPP_DIR}; git checkout ${file})
-            done
-        done
-    fi
+    :
 }
