diff --git a/llm/generate/gen_common.sh b/llm/generate/gen_common.sh
index 0b7b2ae..846cc99 100644
--- a/llm/generate/gen_common.sh
+++ b/llm/generate/gen_common.sh
@@ -51,8 +51,6 @@ git_module_setup() {
         echo "Cleaning up old submodule"
         rm -rf ${LLAMACPP_DIR}
     fi
-    git submodule init
-    git submodule update --force ${LLAMACPP_DIR}
 
 }
 
@@ -64,12 +62,7 @@ apply_patches() {
 
     # apply temporary patches until fix is upstream
     for patch in ../patches/*.diff; do
-        for file in $(grep "^+++ " ${patch} | cut -f2 -d' ' | cut -f2- -d/); do
-            (cd ${LLAMACPP_DIR}; git checkout ${file})
-        done
-    done
-    for patch in ../patches/*.diff; do
-        (cd ${LLAMACPP_DIR} && git apply ${patch})
+        (cd ${LLAMACPP_DIR} && patch -Np1 -i ${patch})
     done
 
     # Avoid duplicate main symbols when we link into the cgo binary
@@ -108,5 +101,5 @@ compress_libs() {
 
 # Keep the local tree clean after we're done with the build
 cleanup() {
-    (cd ${LLAMACPP_DIR}/examples/server/ && git checkout CMakeLists.txt server.cpp)
+    :
 }
