diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0f521b2..061355f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,10 +74,15 @@ set_package_properties(FLEX PROPERTIES PURPOSE "Lexer generator for parsing open
 find_package(BISON REQUIRED)
 set_package_properties(BISON PROPERTIES PURPOSE "Grammar generator for parsing opening hours expressions")
 
-find_package(Python COMPONENTS Development)
-if (Python_FOUND)
+find_package(PythonLibs 3.6 MODULE)
+if (PythonLibs_FOUND)
+    # Extract major/minor python version
+    string(REPLACE "." ";" VERSION_LIST ${PYTHONLIBS_VERSION_STRING})
+    list(GET VERSION_LIST 0 PYTHONLIBS_VERSION_MAJOR)
+    list(GET VERSION_LIST 1 PYTHONLIBS_VERSION_MINOR)
+
     # boost-python requires exact version number
-    find_package(Boost COMPONENTS python${Python_VERSION_MAJOR}${Python_VERSION_MINOR} MODULE)
+    find_package(Boost COMPONENTS python${PYTHONLIBS_VERSION_MAJOR}${PYTHONLIBS_VERSION_MINOR} MODULE)
 endif()
 
 add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050d00)
diff --git a/PyKOpeningHours/CMakeLists.txt b/PyKOpeningHours/CMakeLists.txt
index 6a8e489..49bced3 100644
--- a/PyKOpeningHours/CMakeLists.txt
+++ b/PyKOpeningHours/CMakeLists.txt
@@ -21,8 +21,8 @@ kde_target_enable_exceptions(PyKOpeningHours PRIVATE)
 target_compile_definitions(PyKOpeningHours PRIVATE KOPENINGHOURS_VALIDATOR_ONLY)
 
 # Set up the libraries and header search paths for this target
-target_link_libraries(PyKOpeningHours PUBLIC ${Boost_LIBRARIES} Python::Python KOpeningHours)
-target_include_directories(PyKOpeningHours PRIVATE ${Boost_INCLUDE_DIR})
+target_link_libraries(PyKOpeningHours PUBLIC ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} KOpeningHours)
+target_include_directories(PyKOpeningHours PRIVATE ${Boost_INCLUDE_DIR} ${PYTHON_INCLUDE_DIRS})
 
 set(_install_dir lib/python${PYTHONLIBS_VERSION_MAJOR}.${PYTHONLIBS_VERSION_MINOR}/site-packages)
 
