diff --git a/qt6/src/platformtheme/kioopenwith.cpp b/qt6/src/platformtheme/kioopenwith.cpp
index eff8255..ecc1530 100644
--- a/qt6/src/platformtheme/kioopenwith.cpp
+++ b/qt6/src/platformtheme/kioopenwith.cpp
@@ -8,7 +8,7 @@
 #include <QDBusPendingCall>
 #include <QDBusPendingReply>
 #include <QWidget>
-#include <private/qgenericunixservices_p.h>
+#include <private/qdesktopunixservices_p.h>
 #include <private/qguiapplication_p.h>
 #include <qpa/qplatformintegration.h>
 
@@ -57,7 +57,7 @@ void KIOOpenWith::promptUserForApplication(KJob *job, const QList<QUrl> &urls, c
     if (widget) {
         widget->window()->winId(); // ensure we have a handle so we can export a window (without this windowHandle() may be null)
         auto services = QGuiApplicationPrivate::platformIntegration()->services();
-        if (auto unixServices = dynamic_cast<QGenericUnixServices *>(services)) {
+        if (auto unixServices = dynamic_cast<QDesktopUnixServices *>(services)) {
             windowId = unixServices->portalWindowIdentifier(widget->window()->windowHandle());
         }
     }
