From 601f88f1d05f00e621976be58d170355caa29fdd Mon Sep 17 00:00:00 2001
From: Jonathan Esk-Riddell <jr@jriddell.org>
Date: Fri, 24 Nov 2023 13:25:59 +0000
Subject: [PATCH] used renamed kgantt

---
 CMakeLists.txt     | 4 ++--
 src/CMakeLists.txt | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8e9ab37a..e03f57bb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,8 +53,8 @@ find_package(KF6Holidays ${KF_MIN_VERSION} CONFIG REQUIRED)
 find_package(KF6Contacts ${KF_MIN_VERSION} CONFIG REQUIRED)
 find_package(KF6ConfigWidgets ${KF_MIN_VERSION} CONFIG REQUIRED)
 
-set(KDIAGRAM_LIB_VERSION "1.4.0")
-find_package(KGantt ${KDIAGRAM_LIB_VERSION} CONFIG REQUIRED)
+set(KDIAGRAM_LIB_VERSION "3.0.0")
+find_package(KGantt6 ${KDIAGRAM_LIB_VERSION} CONFIG REQUIRED)
 
 find_package(KPim6Libkdepim ${LIBKDEPIM_LIB_VERSION} CONFIG REQUIRED)
 find_package(KPim6CalendarUtils ${CALENDARUTILS_LIB_VERSION} CONFIG REQUIRED)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5d8917de..7635c7d8 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -132,7 +132,7 @@ PRIVATE
   KF6::I18n
   KPim6::Libkdepim
   KF6::Codecs
-  KGantt
+  KGantt6
   KPim6::AkonadiWidgets
   KF6::Contacts
 )
-- 
GitLab

