diff --git a/CMakeLists.txt b/CMakeLists.txt
index 03790b0..e1810ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,7 +66,7 @@ find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS
     CoreAddons
     Notifications
 )
-find_package(KF6KWeatherCore 0.6.0 REQUIRED)
+find_package(KWeatherCore 0.6.0 REQUIRED)
 
 find_package(KF6KirigamiAddons 0.11 REQUIRED)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e512ddf..1611fe8 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -48,7 +48,7 @@ target_link_libraries(kweather
     KF6::I18n
     KF6::CoreAddons
     KF6::Notifications
-    KF6::KWeatherCore
+    KWeatherCore
     kweatherLib
 )
 
diff --git a/src/plasmoid/CMakeLists.txt b/src/plasmoid/CMakeLists.txt
index 110b27e..c848b54 100644
--- a/src/plasmoid/CMakeLists.txt
+++ b/src/plasmoid/CMakeLists.txt
@@ -18,7 +18,7 @@ add_library(plasma_applet_kweather_1x4 MODULE ${kweather_1x4_SRCS})
 target_link_libraries(plasma_applet_kweather_1x4
                       Qt::Gui
                       Qt::Qml
-                      KF6::KWeatherCore
+                      KWeatherCore
                       KF6::Plasma
                       KF6::I18n
                       kweatherLib)
