diff --git a/src/integration/propertiesplugin/gdrivepropertiesplugin.cpp b/src/integration/propertiesplugin/gdrivepropertiesplugin.cpp
index 4c58ab4..20558d2 100644
--- a/src/integration/propertiesplugin/gdrivepropertiesplugin.cpp
+++ b/src/integration/propertiesplugin/gdrivepropertiesplugin.cpp
@@ -10,7 +10,6 @@
 
 #include <KPluginFactory>
 #include <QClipboard>
-#include <QtGlobal>
 #include <QDesktopServices>
 #include <KIO/StatJob>
 #include <KLocalizedString>
diff --git a/src/integration/propertiesplugin/gdrivepropertiesplugin.h b/src/integration/propertiesplugin/gdrivepropertiesplugin.h
index d36b2cc..407ea55 100644
--- a/src/integration/propertiesplugin/gdrivepropertiesplugin.h
+++ b/src/integration/propertiesplugin/gdrivepropertiesplugin.h
@@ -7,8 +7,13 @@
 #ifndef GDRIVEPROPERTIESPLUGIN_H
 #define GDRIVEPROPERTIESPLUGIN_H
 
+#include <QtGlobal>
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+#include <KPropertiesDialog>
+#else
 #include <KPropertiesDialogPlugin>
 #include <KPropertiesDialog>
+#endif
 
 #include "ui_gdrivepropertiesplugin.h"
 
