diff --git a/CMakeLists.txt b/CMakeLists.txt
index b47591da4..fbc0bd38a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,8 +90,8 @@ endif()
 find_package(KF6 ${KF6_MIN_VERSION} OPTIONAL_COMPONENTS
     DocTools
 )
-find_package(KF6Activities)
-set_package_properties(KF6Activities PROPERTIES DESCRIPTION "KActivities libraries"
+find_package(PlasmaActivities)
+set_package_properties(PlasmaActivities PROPERTIES DESCRIPTION "PlasmaActivities libraries"
                        URL "https://www.kde.org"
                        TYPE OPTIONAL
                        PURPOSE "For tracking which folders are frequently accessed on a Plasma desktop"
@@ -129,8 +129,8 @@ set_package_properties(KF6FileMetaData PROPERTIES
                        PURPOSE "For accessing file metadata labels"
                       )
 
-if (KF6Activities_FOUND)
-    set(HAVE_KACTIVITIES TRUE)
+if (PlasmaActivities_FOUND)
+    set(HAVE_PLASMA_ACTIVITIES TRUE)
 endif()
 
 if (KF6Baloo_FOUND AND KF6BalooWidgets_FOUND AND KF6FileMetaData_FOUND)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index be83cd2d0..533b5d221 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -437,10 +437,10 @@ target_link_libraries(dolphinstatic
     Phonon::phonon4qt6
 )
 
-if (HAVE_KACTIVITIES)
+if (HAVE_PLASMA_ACTIVITIES)
     target_link_libraries(
         dolphinstatic
-        KF6::Activities
+        Plasma::Activities
     )
 endif()
 
diff --git a/src/config-dolphin.h.cmake b/src/config-dolphin.h.cmake
index af78c8622..797ea38c5 100644
--- a/src/config-dolphin.h.cmake
+++ b/src/config-dolphin.h.cmake
@@ -1,5 +1,5 @@
 #cmakedefine01 HAVE_BALOO
-#cmakedefine01 HAVE_KACTIVITIES
+#cmakedefine01 HAVE_PLASMA_ACTIVITIES
 #cmakedefine01 HAVE_KUSERFEEDBACK
 #cmakedefine01 HAVE_PACKAGEKIT
 #cmakedefine01 HAVE_TERMINAL
diff --git a/src/dolphinviewcontainer.cpp b/src/dolphinviewcontainer.cpp
index f5cb459fb..a4db8b74a 100644
--- a/src/dolphinviewcontainer.cpp
+++ b/src/dolphinviewcontainer.cpp
@@ -20,8 +20,8 @@
 #include "statusbar/dolphinstatusbar.h"
 
 #include <KActionCollection>
-#if HAVE_KACTIVITIES
-#include <KActivities/ResourceInstance>
+#if HAVE_PLASMA_ACTIVITIES
+#include <PlasmaActivities/ResourceInstance>
 #endif
 #include <KApplicationTrader>
 #include <KFileItemActions>
@@ -574,7 +574,7 @@ void DolphinViewContainer::setUrl(const QUrl &newUrl)
         m_urlNavigator->setLocationUrl(newUrl);
     }
 
-#if HAVE_KACTIVITIES
+#if HAVE_PLASMA_ACTIVITIES
     KActivities::ResourceInstance::notifyAccessed(newUrl);
 #endif
 }
