diff --git a/core/settings.kcfg b/core/settings.kcfg
index 443c97e2..94da19cb 100644
--- a/core/settings.kcfg
+++ b/core/settings.kcfg
@@ -57,7 +57,7 @@
     <!-- Set by KMixPrefDlg, read by kmix_autostart.desktop	-->
 
     <entry name="AutoStart" type="Bool">
-      <default>true</default>
+      <default>false</default>
     </entry>
 
     <!-- Saved by KMixWindow::saveBaseConfig() and loaded	-->
diff --git a/desktop/kmix_autostart.desktop b/desktop/kmix_autostart.desktop
index 5caa863f..b1f0f8ca 100755
--- a/desktop/kmix_autostart.desktop
+++ b/desktop/kmix_autostart.desktop
@@ -7,7 +7,7 @@ X-KDE-autostart-after=panel
 X-KDE-autostart-after=pulseaudio
 X-KDE-StartupNotify=false
 X-DBUS-StartupType=Unique
-X-KDE-autostart-condition=kmixrc:Global:AutoStart:true
+X-KDE-autostart-condition=kmixrc:Global:AutoStart:false
 Icon=kmix
 GenericName=Sound Mixer
 GenericName[ar]=مازج الصوت
