diff --git a/src/wayland/seat.cpp b/src/wayland/seat.cpp
index b513458c438..d7d84754eb7 100644
--- a/src/wayland/seat.cpp
+++ b/src/wayland/seat.cpp
@@ -188,7 +188,8 @@ void SeatInterface::endDrag()
     AbstractDataSource *dragSource = d->drag.source;
 
     if (dragSource) {
-        if (dragTargetDevice && dragSource->isAccepted() && dragSource->selectedDndAction() != DataDeviceManagerInterface::DnDAction::None) {
+        // TODO: Also check the current drag-and-drop action.
+        if (dragTargetDevice && dragSource->isAccepted()) {
             Q_EMIT dragDropped();
             dragTargetDevice->drop();
             dragSource->dropPerformed();
-- 
GitLab

