diff --git a/src/plugins/qpa/integration.cpp b/src/plugins/qpa/integration.cpp
index d1a5ab9dbe..9aa74c8bb3 100644
--- a/src/plugins/qpa/integration.cpp
+++ b/src/plugins/qpa/integration.cpp
@@ -51,7 +51,7 @@ Integration::Integration()
     , QPlatformIntegration()
     , m_fontDb(new QGenericUnixFontDatabase())
     , m_nativeInterface(new QPlatformNativeInterface())
-    , m_services(new QGenericUnixServices())
+    , m_services(new QDesktopUnixServices())
     , m_clipboard(new Clipboard())
 {
     QWindowSystemInterface::setSynchronousWindowSystemEvents(true);
diff --git a/src/plugins/qpa/integration.h b/src/plugins/qpa/integration.h
index 78a85c8eb0..22e23bae54 100644
--- a/src/plugins/qpa/integration.h
+++ b/src/plugins/qpa/integration.h
@@ -12,7 +12,7 @@
 #include <epoxy/egl.h>
 
 #include <QObject>
-#include <QtGui/private/qgenericunixservices_p.h>
+#include <QtGui/private/qdesktopunixservices_p.h>
 #include <qpa/qplatformintegration.h>
 
 namespace KWin
@@ -61,7 +61,7 @@ private:
     std::unique_ptr<QPlatformNativeInterface> m_nativeInterface;
     QPlatformPlaceholderScreen *m_dummyScreen = nullptr;
     QHash<Output *, Screen *> m_screens;
-    std::unique_ptr<QGenericUnixServices> m_services;
+    std::unique_ptr<QDesktopUnixServices> m_services;
     std::unique_ptr<Clipboard> m_clipboard;
 };
 
